lib/baby-braspag/credit_card.rb

```diff
@@ -186,25 +185,53 @@ def self.just_click_shop(params = {})
       ProtectedCreditCard.just_click_shop(params)
     end
 
-    def self.info_url
-      connection = Braspag::Connection.instance
-      connection.braspag_url + (connection.production? ? PRODUCTION_INFO_URI : HOMOLOGATION_INFO_URI)
-    end
+    class << self
+      private
 
-    def self.authorize_url
-      Braspag::Connection.instance.braspag_url + AUTHORIZE_URI
-    end
+      def check_params(params)
+        super
 
-    def self.capture_url
-      Braspag::Connection.instance.braspag_url + CAPTURE_URI
-    end
+        [:customer_name, :holder, :card_number, :expiration, :security_code, :number_payments, :type].each do |param|
+          raise IncompleteParams if params[param].nil?
+        end
 
-    def self.partial_capture_url
-      Braspag::Connection.instance.braspag_url + PARTIAL_CAPTURE_URI
-    end
+        raise InvalidHolder if params[:holder].to_s.size < 1 || params[:holder].to_s.size > 100
+
+        matches = params[:expiration].to_s.match /^(\d{2})\/(\d{2,4})$/
```

Comment:
It's very unlikely we'd get a string like `12/345`, however, since this is lib code, I'd better be safe than sorry and strictly allow only `xx/xx` and `xx/xxxx`. wdyt?